app/services/page_cloner.rb

```diff
@@ -0,0 +1,79 @@
+class PageCloner
+  def self.clone!(page)
+    new(page).clone!
+  end
+
+  def initialize(page)
+    @page = page
+  end
+
+  def clone!
+    cloned_page = @page.dup
+
+    # TODO: I'm thinking to have the user of this API
+    # decide what the new title should be and have it
+    # passed in to clone! It's not this class' responsibility
+    # to do the necessary checks against AK
```

Comment:
So the check against AK would be done at the point where the campaigner chooses a new title/internal name for the page? The issue atm is that I can imagine campaigners wanting both pages to have the same title. So there's the title vs. internal name distinction to take care of (unless that's already been done).